src/_pytest/pathlib.py

```diff
@@ -64,8 +70,7 @@ def force_writable_and_retry(function, p, excinfo):
 
         chmod_w(p)
         try:
-            # retry the function that failed
-            function(str(p))
+            shutil.rmtree(str(p))
```

Comment:
This needs to be special cased for dir/file (TestNumberedDir.test_rmtree_with_read_only_directory fails).